scrapy/responsetypes.py

```diff
@@ -21,8 +21,8 @@ class ResponseTypes:
         'application/xhtml+xml': 'scrapy.http.HtmlResponse',
         'application/vnd.wap.xhtml+xml': 'scrapy.http.HtmlResponse',
         'application/xml': 'scrapy.http.XmlResponse',
-        'application/json': 'scrapy.http.TextResponse',
-        'application/x-json': 'scrapy.http.TextResponse',
+        'application/json': 'scrapy.http.JsonResponse',
+        'application/x-json': 'scrapy.http.JsonResponse',
```

Comment:
Actually I wonder if it is better to add .json() method to TextResponse, which is contrary to what's been discussed here before. A problem with using HTTP headers is that in a real world not all APIs would return a proper Content-Type header, and .json() method wouldn't be available for them. What's the reason to only providing .json() for responses we're sure are json?